### DISCUSSION TITLE
More font mesurments :)

### DISCUSSION THREAD

**Discussion by Dzivo** [SixLabors/Fonts#151] [Ideas]
Feature request:

Hi guys i want to say first of all great work we have been using your library for some time and it works perfectly.
We have a feature sugestion you might be interested in.

It would be nice if we could align or at least get these mesurments from the `TextMeasurer` class

For me i would like to see Cap Height as well as X-height

![image](https://user-images.githubusercontent.com/5777923/92238679-0006e100-eeba-11ea-97ac-12a9272fefa7.png)

**Comment by Dzivo** [comment]
@antonfirsov thx for the transfers this is maybe even for the Fonts library

**Comment by Dzivo** [comment]
Any news on this ?

**Comment by scalablecory** [comment]
In support of this feature, here's how I'm using the library now and what I'm attempting to do -- I'm trying to match the baseline of different font sizes:

![image-5](https://user-images.githubusercontent.com/4665835/137027060-d800b698-401c-4919-88e9-82a7aa33f18c.png)

Right now I'm just using a "fudge" number to get them to appear on the same line:
https://github.com/scalablecory/aether/blob/6df481eb006a4534c450b12e409655d4bf9aab6e/src/Aether/Devices/Displays/Themes/MultiLineTheme.cs#L92-L94

**Comment by JimBobSquarePants** [comment]
As of 1.0.0-beta16 we've added a ton of metrics to our [FontMetrics](https://docs.sixlabors.com/api/Fonts/SixLabors.Fonts.FontMetrics.html#properties) and [GlyphMetrics](https://docs.sixlabors.com/api/Fonts/SixLabors.Fonts.GlyphMetrics.html#properties) classes. 

These are all given in font units and can be converted to the scaled values using the following formula.
```
pointsize * dpi / metrics.ScaleFactor
```

There's a few more that can be parsed directly from the font but I'd need help finishing things off.

**Comment by AndrewShepherd** [comment]
As somebody new to Fonts, I didn't know what measurements like *AdvanceHeight* or *LeftSideBearing* represented. 

So I wrote a program to illustrate them. Here are some examples:

![lower_case_a](https://user-images.githubusercontent.com/2449034/182828727-7783ad7b-5bea-4124-bc8a-48df6db5318f.png)


The bearing values can actually be negative.

An example of a glpyh with a negative left side bearing:

![lower_case_j](https://user-images.githubusercontent.com/2449034/182828736-c01f2125-cdaf-4411-b49f-c7fef3c6592b.png)

An example of a glyph with a negative right sight bearing:

![upper_case_q](https://user-images.githubusercontent.com/2449034/182828748-903ac734-93ef-41f4-aa88-c99f17819250.png)

**Reply by JimBobSquarePants** [reply]
@AndrewShepherd I don't know how I missed this excellent answer! Do you still have the code for this program? It would be VERY useful.!

**Reply by AndrewShepherd** [reply]
@JimBobSquarePants Sure, happy to share it.

https://github.com/AndrewShepherd/display-font-metrics
